test/Microsoft.Windows.CsWin32.Tests/GeneratorTests.cs

```diff
@@ -108,6 +108,18 @@ public void SimplestMethod(string tfm)
         }
     }
 
+    [Fact]
+    public void SupportedOSPlatform_AppearsOnFriendlyOverloads()
+    {
+        const string methodName = "GetStagedPackagePathByFullName2";
+        this.compilation = this.starterCompilations["net5.0"];
+        this.generator = new Generator(this.metadataStream, DefaultTestGeneratorOptions, this.compilation, this.parseOptions);
+        Assert.True(this.generator.TryGenerate(methodName, CancellationToken.None));
+        this.CollectGeneratedCode(this.generator);
+        this.AssertNoDiagnostics();
+        Assert.All(this.FindGeneratedMethod(methodName), method => Assert.Contains(method.AttributeLists, al => IsAttributePresent(al, "SupportedOSPlatform")));
+    }
```

Comment:
What happens on `netstandard2.0` or `net4.8`? Ideal would be for SupportedOSPlatformAttribute to be defined as internal.

Reply:
The point of this attribute is to allow an analyzer to compare against the TFM, and AFAIK only the net5.0 TFM supports mixing in the Windows version. So even if we did define the attribute ourselves, would the analyzer even be able to do its work?

Reply:
I'm merging the PR because the change you propose is outside the scope of this PR anyway. But if you reply I'll still read it.

Reply:
Ah, I guess you'd have to declare `<TargetPlatformIdentifier>` yourself. Good point, not likely someone would do that and I haven't checked how the .NET 5 SDK analyzers behave in this situation when targeting netcoreapp3.1.

My main concern is whether the generated code is checking for the availability of this attribute in the compilation before emitting attribute syntax though. I didn't spot the check.

Reply:
Ah, found the check:

https://github.com/microsoft/CsWin32/blob/fe62ed485959c67e2ae5f899b8cc316010276af9/src/Microsoft.Windows.CsWin32/Generator.cs#L345

Reply:
And then there's the surprising bit that the attribute isn't allowed on interfaces till .NET 6, which we also handle:

https://github.com/microsoft/CsWin32/blob/fe62ed485959c67e2ae5f899b8cc316010276af9/src/Microsoft.Windows.CsWin32/Generator.cs#L350-L350